codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/generators/BuilderGenerator.kt

```diff
@@ -258,6 +275,42 @@ class BuilderGenerator(
                 withBlock("$memberName: self.$memberName", ",") {
                     // Write the modifier
                     when {
+                        !memberSymbol.isOptional() && member.hasTrait<DefaultTrait>() -> {
```

Comment:
Clients shouldn't set defaults, so this code should be in the server module.

Reply:
Changed, thank you